smallrye-reactive-messaging-jms/src/main/java/io/smallrye/reactive/messaging/jms/JmsConnector.java

```diff
@@ -0,0 +1,150 @@
+package io.smallrye.reactive.messaging.jms;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.*;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.literal.NamedLiteral;
+import javax.inject.Inject;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSContext;
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.spi.Connector;
+import org.eclipse.microprofile.reactive.messaging.spi.IncomingConnectorFactory;
+import org.eclipse.microprofile.reactive.messaging.spi.OutgoingConnectorFactory;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
+
+@ApplicationScoped
+@Connector(JmsConnector.CONNECTOR_NAME)
+public class JmsConnector implements IncomingConnectorFactory, OutgoingConnectorFactory {
+
+    static final String CONNECTOR_NAME = "smallrye-jms";
+
+    @Inject
+    Instance<ConnectionFactory> factories;
+
+    @Inject
+    Instance<Jsonb> jsonb;
+
+    @Inject
+    @ConfigProperty(name = "smallrye.jms.threads.max-pool-size", defaultValue = "10")
```

Comment:
Do you think it makes sense to put the constants in an API (such as Interface) so that it can be viewed and changed easier in the future?

Reply:
Not necessary in an interface, but I've extracted the values into constant in the JMSConnector class.

Reply:
Done.